pennylane/tape/tapes/tape.py

```diff
@@ -96,6 +97,31 @@ def expand_tape(tape, depth=1, stop_at=None, expand_measurements=False):
 
     new_tape = tape.__class__()
 
+    if expand_measurements:
+        # Check for observables acting on the same wire. If present, observables must be
+        # qubit-wise commuting Pauli words. In this case, the tape is expanded with joint
+        # rotations and the observables updated to the computational basis. Note that this
+        # expansion acts on the original tape in place.
+
+        obs = [m.obs for m in tape.measurements if m.obs is not None]
+        m_idx = [i for i, m in enumerate(tape.measurements) if m.obs is not None]
+        obs_wires = [wire for m in tape.measurements for wire in m.wires if m.obs is not None]
+
+        if len(obs_wires) != len(set(obs_wires)):
+            try:
+                rotations, diag_obs = diagonalize_qwc_pauli_words(obs)
+            except ValueError as e:
```

Comment:
I was 50/50 on whether to re-raise. The base error is "X and Y are not qubit-wise commuting". This might be obvious to some, but to others maybe it's better to clarify that we are talking about observables on the same wire.

Reply:
Perhaps this is a sign that the error message from `diagonalize_qwc_pauli_words` could be improved?

But this is minor. As much as it would be nice to rely on existing validation, if we need to add additional info, I suppose there is nothing much we can do apart from re-raising.